RELEASES.md

```diff
@@ -205,3 +205,93 @@ To create a minor release:
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
    - Remove all `R:minor` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+
+## Minor Release Checklist
+
+The following set of steps are performed on all releases that increment
+the _minor_ version. These steps ensure that Tendermint is well tested, stable,
+and suitable for adoption by the various diverse projects that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters a two week
+'Feature Freeze'. A feature freeze means that _no_ new features or functionality
+are added to the code base. The Tendermint team instead directs all attention to
+ensuring that the existing code is stable and reliable. Broken tests are fixed,
+flakey-tests are remedied, end-to-end test failures are thoroughly diagnosed and
+all efforts of the team are aimed at improving the quality of the code. During
+this period, the upgrade harness tests are run repeatedly and a variety of
+in-house testnets are run to ensure Tendermint functions at the scale it will
+be used by application developers and node operators.
+
+### Nightly End-To-End Tests
+
+The Tendermint team maintains a set of end-to-end tests that run each night
+on the latest commit of the project. During the feature freeze, these tests are
+run nightly and must pass consistently for a release of Tendermint to be
+considered stable.
```

Comment:
If we're going to have a sentence or two in the other sections as to what these processes set to achieve or how they lead to an improved quality of release then we should also have one here. We can't assume that people reading this have the same degree of context that we do

Reply:
Can we maybe link to the e2e documentation for more information

Reply:
Linked and added.